src/docs/MLOS/MLOSUserGuide.adoc

```diff
@@ -166,6 +166,426 @@ E. Click the *Tasks* tab, select an *Preview_Results* task and click on the *Pre
+
+```python
+#%selection_script(name=TASK_NAME)
+selected = True
+```
+
+A second way is by providing the name of the task, and the path of a .py file containing the selection code:
+
+```python
+#%selection_script(name=TASK_NAME, path=PATH_TO/SELECTION_CODE_FILE.py)
+```
+
+==== Adding job fork environment and/or selection script
+
+If the selection scripts and/or the fork environments are the same for all job tasks, we can add them just once using
+the `job_selection_script` and/or the `job_fork_env` pragmas.
+
+Usage:
+
+For a job selection script, please use:
+
+```python
+#%job_selection_script([language=SCRIPT_LANGUAGE], [path=./SELECTION_CODE_FILE.py], [force=on/off])
+```
+
+For a job fork environment, use:
+
+```python
+#%job_fork_env([language=SCRIPT_LANGUAGE], [path=./FORK_ENV_FILE.py], [force=on/off])
+```
+
+The `force` parameter says if the pragma has to overwrite the task selection scripts or fork environment already set or not.
+
+==== Adding pre and/or post scripts
+
+Sometimes, specified scripts has to be executed before and/or after a particular task. To do that, the solution provides
+ `pre_script` and `post_script` pragmas.
+
+To add a pre-script to a task, please use:
+
+```python
+#%pre_script(name=TASK_NAME, language=SCRIPT_LANGUAGE, [path=./PRE_SCRIPT_FILE.py])
+```
+
+To add a post-script to a task, use:
+
+```python
+#%post_script(name=TASK_NAME, language=SCRIPT_LANGUAGE, [path=./POST_SCRIPT_FILE.py])
+```
+
+==== Create a job
+
+To create a job, use the `#%job()` pragma:
+
+```python
+#%job(name=JOB_NAME)
+```
+
+If the job has already been created, the call of this pragma would just rename the job already created by the new provided name.
+
+**Notice that it is not necessary to create and name explicitly the job. If not done by the user, this step is implicitly
+performed when the job is submitted (check section <<Submit your job to the scheduler>> for more information).**
+
+==== Visualize job
+
+To visualize the created workflow, use the `#%draw_job()` pragma to plot the graph that represents the job into a separate window:
+
+```python
+#%draw_job()
+```
+
+Two optional parameters can be used to configure the way the kernel plots the workflow graph.
+
+**inline plotting:**
+
+If this parameter is set to 'off', the workflow plotting is made through the https://matplotlib.org/[Matplotlib]
+external window. The default value is 'on'.
+
+```python
+#%draw_job(inline=off)
+```
+
+**saving into hard disk:**
+
+To be sure the workflow is saved into a _.png_ file, this option needs to be set to *on*. The default value is *off*.
+
+```python
+#%draw_job(save=on)
+```
+
+Note that the job's name can take one of the following possible values:
+
+1. The parameter `name`'s value, if provided
+2. The job's name, if created
+3. The notebook's name, if the kernel can retrieve it
+4. `Unnamed_job`
+
+General usage:
+
+```python
+#%draw_job([name=JOB_NAME], [inline=off], [save=on])
+```
+
+==== Save workflow in dot format
```

Comment:
Save the workflow in dot format 

and why not using the same format above Save the workflow locally

Reply:
Saving the graph locally is meannt to be as a picture .png ... etc
.dot is a common format that is related to graphviz. The graph is described as a list of edges/arcs
A1 -> A2
A3 -> A4 
... etc
https://fr.wikipedia.org/wiki/DOT_(langage)

Reply:
-Save the workflow locally
-Save the workflow in dot format